sympy/ntheory/generate.py

```diff
@@ -39,16 +44,32 @@ class Sieve:
     """
 
     # data shared (and updated) by all Sieve instances
-    _list = _array('l', [2, 3, 5, 7, 11, 13])
+    def __init__(self):
+        self._n = 6
+        self._list = _aset(2, 3, 5, 7, 11, 13) # primes
+        self._tlist = _aset(0, 1, 1, 2, 2, 4) # totient
+        self._mlist = _aset(0, 1, -1, -1, 0, -1) # mobius
 
```

Comment:
`assert all(len(i) == self._n for i in (self._list, self._tlist, self._mlist))`

Reply:
Committed to that.
@smichr @jksuom Thank you for the review.